hardware/drone/camera/camera_controller.py

```
@@ -0,0 +1,151 @@
+                temp_path = "/tmp/drone_frame.jpg"
+                cv2.imwrite(temp_path, frame)
+                
+                # Emit the file path
+                self.frameReady.emit(f"file://{temp_path}")
+                
+            if frame is None:
+                return
+
+            # Convert BGR → RGB for proper display
+            frame_rgb = cv2.cvtColor(frame, cv2.COLOR_BGR2RGB)
+            temp_path = pathlib.Path("/tmp/drone_frame.jpg")
```

Ruff (0.14.2)

Probable insecure usage of temporary file or directory: "/tmp/drone_frame.jpg"

(S108)